### PR TITLE
Fix get_all_project_names to use job projection.

### LINKED ISSUE #1094: get_all_project_names doesn't include projects for which there is no testcases
E.g. on fuzzer stats page, the project filter only includes projects for which there is at least one Testcase.

### PR DESCRIPTION
**Pull request by inferno-chromium**
Fixes #1094.

### CODE REVIEW

File: src/python/datastore/data_handler.py [hunk 1]
```diff
@@ -94,10 +94,9 @@
 @memoize.wrap(memoize.Memcache(MEMCACHE_TTL_IN_SECONDS))
 def get_all_project_names():
   """Return all project names."""
-  query = data_types.Testcase.query(
-      projection=[data_types.Testcase.project_name],
-      distinct=True).order(data_types.Testcase.project_name)
-  return [testcase.project_name for testcase in query if testcase.project_name]
+  query = data_types.Job.query(
+      projection=[data_types.Job.project], distinct=True)
+  return sorted([_.project for _ in query])
```
Comment (oliverchang) [inline src/python/datastore/data_handler.py]: nit: s/_/job/ to be clearer?

Comment (inferno-chromium) [reply src/python/datastore/data_handler.py]: Done!

**Pull request merged**